Assets/_BForBoss/_Core/Scripts/Boss/DerekContextManager.cs

```diff
@@ -0,0 +1,175 @@
+        
+        private RingLaborManager _ringLaborManager;
+        private DerekBossManager _bossManager;
+        //private Interactable _TutorialEndButton;
+
+        private WaitForSeconds _LaborIncompleteWaitForSeconds;
+        
+        public enum Phase
+        {
+            Tutorial, // Battle hasn't begun just yet
+            FirstPhase, // Health between 75% and 100%
+            SecondPhase, // Health between 25% and 75%
+            FinalPhase, // Health between 0% and 25%
+            Death, // Derek is dead
+        }
+    
+        //Put this in Boss Manager
+        public enum Vulnerability
+        {
+            Invulnerable, //Shields are up, impervious to damage
+            Vulnerable, //Shields are down, susceptible to damage
+        }
+
+        private Phase _currentPhase = Phase.Tutorial;
+        private Vulnerability _currentVulnerability = Vulnerability.Invulnerable;
+
+#if UNITY_EDITOR //Debug Buttons for WIP PR
+        
+        private RingLaborManager debug_rlm;
+        private DerekBossManager debug_dbm;
+        private Phase debug_currentPhase = Phase.Tutorial;
+        private Vulnerability debug_currentVulnerability = Vulnerability.Invulnerable;
+        
+        private void Debug_Initialize()
+        {
+            debug_rlm = FindObjectOfType<RingLaborManager>();
+            debug_dbm = FindObjectOfType<DerekBossManager>();
+            
+            Initialize(debug_rlm, debug_dbm);
+        }
+
+        [Button]
+        private void Debug_Reset()
+        {
+            debug_currentPhase = Phase.Tutorial;
+            debug_currentVulnerability = Vulnerability.Invulnerable;
+        }
+
+        [Button]
+        private void Debug_AdvancePhase()
+        {
+            if (debug_rlm == null || debug_dbm == null)
+            {
+                Debug_Initialize();
+            }
+
+            debug_currentPhase++;
+            debug_dbm.UpdatePhase(debug_currentPhase);
+            
+            if (debug_currentPhase == Phase.Death)
+            {
+                Debug_Reset();
+            }
+        }
+
+        [Button]
+        private void Debug_SwitchVulnerability()
+        {
+            if (debug_rlm == null || debug_dbm == null)
+            {
+                Debug_Initialize();
+            }
+
+            switch (debug_currentVulnerability)
+            {
+                case Vulnerability.Invulnerable:
+                    debug_currentVulnerability = Vulnerability.Vulnerable;
+                    break;
+                case Vulnerability.Vulnerable:
+                    debug_currentVulnerability = Vulnerability.Invulnerable;
+                    break;
+            }
+            
+            debug_dbm.UpdateVulnerability(debug_currentVulnerability);
+        }
+#endif
+        
+        
+        public void Reset()
+        {
+            if (_ringLaborManager == null || _bossManager == null)
+            {
+                return;
```

Comment:
SInce this class requires `_ringLaborManager` and `_bossManager` . you can use the Panichelper here and crash the game.

And then remove all the `null` checks here.

We don't want to suppress/ignore the problem, we want to be able to find it asap and handle it.

Reply:
Would it be better to have it in the `Initialize` method then, rather than on `Reset`?

Reply:
Wherever it gets initialized once, you can put it there.